src/functions/Email.js

```diff
@@ -1,17 +1,55 @@
-const URL = "https://usvotes-3ulcxuufea-uw.a.run.app/email/";
+import { getUserDatabase, getChallengerDatabase } from "./UserData";
 
-const stopEmail = true; // make false when testing emails
+const URL = "https://usvotes-6vsnwycl4q-uw.a.run.app/email/";
 
-export function emailUser(email, emailType) {
-  let formData = new FormData();
-  formData.append("email", email);
-  formData.append("type", emailType);
+const stopEmail = false; // make false when testing emails
 
+export async function emailUser(email, emailType) {
   if (stopEmail) {
     console.log("no email sent to prevent reaching daily limit while testing");
     return;
   }
 
+  let formData = new FormData();
+  formData.append("email", email);
+  formData.append("type", emailType);
+
+  const userData = await getUserDatabase();
+
+  // emails challenger when badge earned
+  if (emailType === "badgeEarned") {
+    const challengerData = await getChallengerDatabase();
+    let daysLeft = challengerData.challengeEndDate;
+    let milisecondsLeft = new Date(daysLeft.seconds * 1000) - new Date();
+    let days = Math.floor(milisecondsLeft / 1000 / 60 / 60 / 24 + 1);
+    if (days < 1) {
+      days = 0;
+    }
+
+    let badgeArr = challengerData.badges;
+    while (badgeArr.length > 8) {
+      badgeArr.pop();
+    }
+    let badgesLeft = 8 - badgeArr.length;
+
+    formData.append("avatar", userData.avatar);
+    formData.append("daysLeft", days);
+    formData.append("badgesLeft", badgesLeft);
+  }
+
+  // emails challenger when 8 badges earned
+  if (emailType === "challengeWon") {
+    formData.append("avatar", userData.avatar);
+  }
+
+  // emails player
+  if (emailType === "registered" || emailType === "electionReminder") {
+    const challengerData = await getChallengerDatabase();
+
+    formData.append("firstName", challengerData.name);
+    formData.append("avatar", challengerData.avatar);
+  }
```

Comment:
nit: If these cases are mutually exclusive, I would separate them into maybe this:
```
if(...) {

} else if (...) {

} else {
// handle wrong input
} 
```